src/field_effect.c

```diff
@@ -2537,6 +2537,213 @@ static void TeleportInFieldEffectTask3(struct Task *task)
     }
 }
 
+// new functions for Retreat field effect
+static void Task_DoRetreatFieldEffect(u8 taskId);
+static void RetreatFieldEffectTask1(struct Task *task);
+static void RetreatFieldEffectTask2(struct Task *task);
+static void RetreatFieldEffectTask3(struct Task *task);
+static void RetreatFieldEffectTask4(struct Task *task);
+static void FieldCallback_RetreatIn(void);
+static void Task_DoRetreatInFieldEffect(u8 taskId);
+static void RetreatInFieldEffectTask1(struct Task *task);
+static void RetreatInFieldEffectTask2(struct Task *task);
+static void RetreatInFieldEffectTask3(struct Task *task);
+
+static void (*const sRetreatEffectFuncs[])(struct Task *) = {
+    RetreatFieldEffectTask1,
+    RetreatFieldEffectTask2,
+    RetreatFieldEffectTask3,
+    RetreatFieldEffectTask4
+};
+
+void CreateRetreatFieldEffectTask(void)
+{
+    CreateTask(Task_DoRetreatFieldEffect, 0);
+}
+
+static void Task_DoRetreatFieldEffect(u8 taskId)
+{
+    sRetreatEffectFuncs[gTasks[taskId].data[0]](&gTasks[taskId]);
+}
+
+static void RetreatFieldEffectTask1(struct Task *task)
+{
+    LockPlayerFieldControls();
+    FreezeObjectEvents();
+    CameraObjectReset2();
+    task->data[15] = GetPlayerFacingDirection();
+    task->data[0]++;
+}
+
+static void RetreatFieldEffectTask2(struct Task *task)
```

Comment:
I think the functions here could gain an immense amount of clarity by using some defined task data constants, like this snippet from `field_player_avatar.c` for example:
```c
#define tState data[0]
#define tRotationTimer data[1]
#define tDeltaY data[2]
#define tYdeflection data[3]
```

Potentially also some regular preproc constants as well.